compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -2957,13 +2954,17 @@ impl<'a> Parser<'a> {
         err
     }
 
-    pub fn is_diff_marker(&mut self, long_kind: &TokenKind, short_kind: &TokenKind) -> bool {
+    /// This checks if this is a git diff marker, one of
+    /// >>>>>
+    /// =====
+    /// <<<<<
+    pub fn is_git_diff_marker(&mut self, long_kind: &TokenKind, short_kind: &TokenKind) -> bool {
         (0..3).all(|i| self.look_ahead(i, |tok| tok == long_kind))
             && self.look_ahead(3, |tok| tok == short_kind)
     }
 
-    fn diff_marker(&mut self, long_kind: &TokenKind, short_kind: &TokenKind) -> Option<Span> {
-        if self.is_diff_marker(long_kind, short_kind) {
+    fn git_diff_marker(&mut self, long_kind: &TokenKind, short_kind: &TokenKind) -> Option<Span> {
```

Comment:
If I remember correctly it was intentional to not mention `git` here. I'd need to look for the original PRs. CC @estebank

Reply:
Indeed, diff markers are used by git, but not exclusive to it.